src/fsharp/ConstraintSolver.fs

```diff
@@ -648,15 +648,15 @@ let CheckWarnIfRigid (csenv:ConstraintSolverEnv) ty1 (r:Typar) ty =
 /// Propagate all effects of adding this constraint, e.g. to solve other variables 
 let rec SolveTyparEqualsTyp (csenv:ConstraintSolverEnv) ndeep m2 trace ty1 ty =
     let m = csenv.m
-    let denv = csenv.DisplayEnv
+    
     DepthCheck ndeep m  ++ (fun () -> 
     match ty1 with 
     | TType_var r | TType_measure (MeasureVar r) ->
       // The types may still be equivalent due to abbreviations, which we are trying not to eliminate 
       if typeEquiv csenv.g ty1 ty then CompleteD else
 
       // The famous 'occursCheck' check to catch things like 'a = list<'a> 
-      if occursCheck csenv.g r ty then ErrorD (ConstraintSolverInfiniteTypes(denv,ty1,ty,m,m2)) else
+      if occursCheck csenv.g r ty then ErrorD (ConstraintSolverInfiniteTypes(csenv.DisplayEnv,ty1,ty,m,m2)) else
```

Comment:
We may reindent and space the tuple

```fsharp
if occursCheck csenv.g r ty then
    ErrorD (ConstraintSolverInfiniteTypes(csenv.DisplayEnv, ty1, ty, m, m2))
else
// keep same indentation there, that fun () is huge
```

or at least keep the else on the next line

@dsyme what is up with style of spacing in tuples? I've seen some fortran codebase where those won't have any spaces between each argument in the list, I think it reads better with spacing but not sure what you prefer for the compiler codebase.